index.js

```diff
@@ -7,13 +7,26 @@ const toArray = (val) => Array.isArray(val) ? val : [val].filter(Boolean);
 const copyJsRule = (config, options, ruleOptions) => {
     const { dir, config: { distDir } } = options;
 
-    const jsRuleIndex = config.module.rules.findIndex((rule) => rule.use && /\bnext-babel-loader\b/.test(rule.use.loader));
+    const jsRuleIndex = config.module.rules.findIndex((rule) => {
+        if (!rule.use) {
+            return -1;
+        }
+
+        if (typeof rule.use.loader === 'string') {
+            return /\bnext-babel-loader\b/.test(rule.use.loader);
+        }
+
+        return rule.use.findIndex((use) => /\bnext-babel-loader\b/.test(use.loader));
+    });
+
     const jsRule = config.module.rules[jsRuleIndex];
 
     if (!jsRule) {
         throw new Error('Could not find JS rule with next-babel-loader');
     }
 
+    const jsRuleUse = Array.isArray(jsRule.use) ? jsRule.use[1] : jsRule.use;
```

Comment:
We can't assume the babel loader rule is in index 1.

Ideally, we would use the new thread optimization that next.js added, however, that requires changing the cache-loader config, such as the directory. Let's do that in a separate PR, so can you create a new issue for that?

I will explain to you personally how to avoid using an hardcoded index.

Reply:
Solved according to the discussion we had.